packages/gatsby-plugin-offline/src/gatsby-browser.js

```diff
@@ -1,5 +1,6 @@
 exports.registerServiceWorker = () => true
 
+const ignoredLinkRels = /\b(preconnect|prefetch|prerender|dns-prefetch)\b/
```

Comment:
Service workers are lazily being initiate so we don't know if the resource got cached by the service worker, so that's why we re-fetch all possible link tags. If we won't do that our page would be broken as not all resources would be cached.

The reason why I want to keep prefetch is because it's a valid resource unlike preconnect & dns-prefetch. With prerender you actually want to load the "next" or a very likely "next" page so I don't see a big problem with fetching that one.

Reply:
You were right, we should only do preload, stylesheet & scripts not all the rest.